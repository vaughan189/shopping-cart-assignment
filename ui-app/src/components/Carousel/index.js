import React from "react";
import Carousel from "react-material-ui-carousel";
import { CardActionArea } from "@mui/material";
import Card from "@mui/material/Card";
import CardMedia from "@mui/material/CardMedia";

import "./index.scss";

export const Carousal = (props) => {
  const { bannerData } = props;

  return (
    <Carousel className="CarouselItem">
      {bannerData.map((item, i) => (
        <CarouselItem
          key={i}
          item={item}
          bannerImageAlt={item.bannerImageAlt}
          bannerImageUrl={item.bannerImageUrl}
        />
      ))}
    </Carousel>
  );
};

export const CarouselItem = (props) => {
  const { bannerImageUrl, bannerImageAlt } = props;
  return (
    <Card>
      <CardActionArea>
        <CardMedia
          component="img"
          height="400"
          image={bannerImageUrl}
          alt={bannerImageAlt}
        />
      </CardActionArea>
    </Card>
  );
};
