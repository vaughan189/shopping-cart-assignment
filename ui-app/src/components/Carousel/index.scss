.CarouselItem {
  width: 100%;
  margin-top: 20px;
}

img {
  max-width: 100%;
  height: auto;
  padding: 0;
  margin: 0;
}
