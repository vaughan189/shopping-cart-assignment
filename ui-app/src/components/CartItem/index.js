import React from "react";

import {
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  IconButton,
  Typography,
} from "@mui/material";

import AddSharpIcon from "@mui/icons-material/AddSharp";
import RemoveSharpIcon from "@mui/icons-material/RemoveSharp";
import { Box } from "@mui/system";

export const CartItem = (props) => {
  const { cartItem, deleteCartItem, addProductsToCart } = props;
  return (
    <Card sx={{ margin: 1 }}>
      <CardActionArea>
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-between",
          }}
        >
          <CardMedia
            sx={{
              width: { xl: 100, md: 50, sm: 100, xs: 100 },
              height: { xl: 100, md: 50, sm: 100, xs: 100 },
            }}
            component="img"
            height="20"
            image={cartItem.imageURL}
            alt="green iguana"
          />
          <CardContent
            sx={{ width: { xl: "100%", md: "100%", sm: "100%", xs: "100%" } }}
          >
            <Box
              sx={{
                display: "flex",
                flexDirection: "column",
              }}
            >
              <Typography
                gutterBottom
                variant="h6"
                component="div"
                sx={{
                  fontWeight: "Bold",
                  fontSize: {
                    xl: "1rem",
                    md: "1rem",
                    sm: "1rem",
                    xs: "1rem",
                  },
                }}
              >
                {cartItem.productName}
              </Typography>
              <Box
                sx={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "space-between",
                  alignItems: "center",
                }}
              >
                <Box
                  sx={{
                    display: "flex",
                    flexDirection: "row",
                    justifyContent: "space-evenly",
                    alignItems: "flex-start",
                    width: { xl: "40%", md: "40%", sm: "40%", xs: "70%" },
                  }}
                >
                  <IconButton
                    id="subtract-Cart-Item"
                    aria-label="subtract"
                    size="small"
                    sx={{
                      height: "20px",
                      width: "20px",
                      borderRadius: { xl: 5, md: 5, sm: 0, xs: 0 },
                      color: "#FFFFFF",
                      backgroundColor: "#c73f6d",
                      "&.MuiButtonBase-root:hover": {
                        bgcolor: "#c73f6d",
                      },
                    }}
                    onClick={() => deleteCartItem(cartItem.id)}
                  >
                    <RemoveSharpIcon />
                  </IconButton>
                  <Typography
                    sx={{ width: "10px", fontFamily: "Dosis" }}
                    component="div"
                  >
                    {cartItem.quantity}
                  </Typography>
                  <IconButton
                    id="add-Cart-Item"
                    aria-label="add"
                    size="small"
                    sx={{
                      height: "20px",
                      width: "20px",
                      borderRadius: { xl: 5, md: 5, sm: 0, xs: 0 },
                      color: "#FFFFFF",
                      backgroundColor: "#c73f6d",
                      "&.MuiButtonBase-root:hover": {
                        bgcolor: "#c73f6d",
                      },
                    }}
                    onClick={() => addProductsToCart(cartItem)}
                  >
                    <AddSharpIcon />
                  </IconButton>
                  <Typography component="div" sx={{ fontFamily: "Dosis" }}>
                    X
                  </Typography>
                  <Typography component="div" sx={{ fontFamily: "Dosis" }}>
                    Rs.{cartItem.price}
                  </Typography>
                </Box>
                <Box
                  sx={{
                    display: "flex",
                    flexDirection: "row",
                    justifyContent: "space-around",
                    alignItems: "center",
                  }}
                >
                  <Typography component="div" sx={{ fontFamily: "Dosis" }}>
                    Rs. {cartItem.price * cartItem.quantity}
                  </Typography>
                </Box>
              </Box>
            </Box>
          </CardContent>
        </Box>
      </CardActionArea>
    </Card>
  );
};
