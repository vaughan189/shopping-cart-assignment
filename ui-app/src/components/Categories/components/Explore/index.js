import React from "react";
import { Button, Grid } from "@mui/material";
import { useHistory } from "react-router-dom";

import "./index.scss";

export const Explore = (props) => {
  const { name, description } = props;
  let history = useHistory();

  return (
    <Grid
      container
      direction="column"
      justifyContent="center"
      alignItems="center"
      spacing={3}
    >
      <Grid item xs={12} sm={12} md={12} xl={12} className="categoryName">
        {name}
      </Grid>
      <Grid item xs={12} sm={12} md={12} xl={12}>
        {description}
      </Grid>
      <Grid item xs={12} sm={12} md={12} xl={12}>
        <Button
          id="explore-more-btn"
          variant="contained"
          color="primary"
          style={{ textTransform: "none" }}
          onClick={() => history.push("/products")}
        >
          Explore {name}
        </Button>
      </Grid>
    </Grid>
  );
};
