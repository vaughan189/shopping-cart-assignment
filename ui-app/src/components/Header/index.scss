.header {
  background-color: "#FFFFFF";
  height: "10%";
}

img {
  max-width: 80%;
  height: 40%;
  padding: 0;
  margin: 0;
}
