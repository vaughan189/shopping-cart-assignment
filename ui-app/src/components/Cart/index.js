/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from "react";

import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import { CartItem } from "../CartItem";
import { Button, IconButton } from "@mui/material";
import { LowestPriceSection } from "../LowestPrice";

import ArrowForwardIosSharpIcon from "@mui/icons-material/ArrowForwardIosSharp";
import useCart from "../../hooks/useCart";

export const Cart = () => {
  const { cartProductsData, cartProductList, cartProductsDataLoader } =
    useCart();
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
        fontFamily: "Dosis",
      }}
    >
      <CartHeader />
      {!cartProductsDataLoader && (
        <>
          {cartProductList.map((item) => {
            return (
              <Box key={item.id}>
                <CartItem cartItem={item} />
              </Box>
            );
          })}
        </>
      )}
      <LowestPriceSection />
      <CheckOutSection cartProductsData={cartProductsData} />
    </Box>
  );
};

const CartHeader = () => {
  return (
    <Typography
      component="div"
      sx={{
        fontSize: 18,
        height: "40px",
        paddingTop: 2,
        paddingLeft: 2,
        fontWeight: "bold",
        backgroundColor: {
          xl: "#000000",
          md: "#000000",
          sm: "#FFFFFF",
          xs: "#FFFFFF",
        },
        color: {
          xl: "#FFFFFF",
          md: "#FFFFFF",
          sm: "#000000",
          xs: "#000000",
        },
        fontFamily: "Dosis",
      }}
    >
      My Cart{" "}
      <Typography
        component="span"
        sx={{
          fontFamily: "Dosis",
          fontSize: 16,
        }}
      >
        (1 item)
      </Typography>
    </Typography>
  );
};

const CheckOutSection = (props) => {
  const { cartProductsData } = props;
  const [totalPrice, setTotalPrice] = useState(0);

  const getTotalPrice = () => {
    return cartProductsData.data.reduce(
      (a, b) => {
        return {
          price: a.price + b.price,
        };
      },
      {
        price: 0,
      }
    );
  };

  useEffect(() => {
    if (
      cartProductsData &&
      cartProductsData.data &&
      cartProductsData.data.length > 0
    ) {
      setTotalPrice(getTotalPrice());
    }
  }, [cartProductsData]);

  return (
    <Box
      sx={{
        bottom: 0,
        display: "flex",
        flexDirection: "column",
        justifyContent: "centre",
        alignItems: "centre",
        marginTop: { xl: "70%", md: "70%", sm: "88%", xs: "83%" },
        backgroundColor: "white",
        padding: 2,
      }}
    >
      <Typography
        sx={{
          display: "flex",
          flexDirection: "row",
          justifyContent: "center",
          fontFamily: "Dosis",
          fontSize: 16,
          fontWeight: 600,
        }}
      >
        Promo code can be applied on payment page
      </Typography>
      <Button
        variant="contained"
        sx={{
          display: "flex",
          flexDirection: "row",
          justifyContent: "space-between",
          alignItems: "center",
        }}
      >
        <Typography component="span">Proceed to Checkout</Typography>
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-around",
            alignItems: "center",
          }}
        >
          <Typography component="span" sx={{ textTransform: "none" }}>
            Rs.{totalPrice.price}
          </Typography>
          <IconButton aria-label="delete" size="small" color="secondary">
            <ArrowForwardIosSharpIcon
              fontSize="inherit"
              sx={{ color: "#FFFFFF" }}
            />
          </IconButton>
        </Box>
      </Button>
    </Box>
  );
};
