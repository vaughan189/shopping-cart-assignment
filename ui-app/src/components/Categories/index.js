import React from "react";
import { Card, CardMedia, Grid } from "@mui/material";

import "./index.scss";
import { Explore } from "./components/Explore";

export const Categories = (props) => {
  const { categoriesData } = props;
  return (
    <>
      {categoriesData.map((item, index) => {
        return (
          <CategoriesItem
            key={item.id}
            name={item.name}
            description={item.description}
            imageUrl={item.imageUrl}
            position={index % 2}
          />
        );
      })}
    </>
  );
};

export const CategoriesItem = (props) => {
  const { name, description, imageUrl, position } = props;
  return (
    <Card className="CategoriesItem" sx={{ minHeight: 260 }}>
      <Grid
        container
        direction="row"
        justifyContent="center"
        alignItems="center"
      >
        {position === 0 && (
          <>
            <Grid item xs={4} sm={4} md={4} xl={4}>
              <CardMedia
                rel="preload"
                component="img"
                height="400"
                image={imageUrl}
                alt={name}
              />
            </Grid>
            <Grid item xs={6} sm={6} md={6} xl={6}>
              <Explore name={name} description={description} />
            </Grid>
          </>
        )}
        {position === 1 && (
          <>
            <Grid item xs={6} sm={6} md={6} xl={6}>
              <Explore name={name} description={description} />
            </Grid>
            <Grid item xs={4} sm={4} md={4} xl={4}>
              <CardMedia
                rel="preload"
                component="img"
                height="400"
                image={imageUrl}
                alt={name}
              />
            </Grid>
          </>
        )}
      </Grid>
    </Card>
  );
};
