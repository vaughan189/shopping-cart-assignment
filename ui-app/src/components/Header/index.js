/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from "react";
import { useHistory } from "react-router-dom";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Container from "@mui/material/Container";
import Button from "@mui/material/Button";
import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
import {
  ButtonGroup,
  Grid,
  Popover,
  Typography,
  useMediaQuery,
} from "@mui/material";

import { Cart } from "../Cart";
import useCart from "../../hooks/useCart";
import { cartService } from "../../services";

import "./index.scss";

const routes = [
  {
    id: 1,
    name: "Home",
    path: "/home",
  },
  {
    id: 2,
    name: "Products",
    path: "/products",
  },
];

export const Header = () => {
  let history = useHistory();

  const screenIsSM = useMediaQuery(
    "@media (min-width:600px) and (max-width:899.95px)"
  );
  const screenIsXS = useMediaQuery(
    "@media (min-width:0px) and (max-width:599.95px)"
  );

  const { cartProductsData, getCartProducts } = useCart();

  const [anchorEl, setAnchorEl] = useState(null);

  const handleClick = (event) => {
    if (screenIsSM || screenIsXS) {
      history.push("/cart");
    } else setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const open = Boolean(anchorEl);
  const id = open ? "simple-popover" : undefined;

  const redirectToPage = (path) => {
    history.push(path);
  };

  useEffect(() => {
    const subscription = cartService.getCartData().subscribe((data) => {
      if (data) {
        getCartProducts();
      }
    });
    return () => {
      subscription.unsubscribe();
    };
  }, []);

  return (
    <AppBar
      position="sticky"
      style={{ backgroundColor: "#FFFFFF", height: "10%" }}
    >
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <Grid container direction="row">
            <Grid item xl={2} md={2} sm={2} xs={2}>
              <img
                id="home-icon-btn"
                src={require("../../assets/logo.png")}
                alt="Sabka Bazaar Logo"
                onClick={() => {
                  history.push("/home");
                }}
                style={{ cursor: "pointer" }}
              />
            </Grid>
            <Grid item xl={6} md={6} sm={6} xs="auto">
              <Grid
                container
                direction="row"
                justifyContent="center"
                alignItems="center"
              >
                <Box
                  sx={{
                    display: {
                      xs: "none",
                      md: "contents",
                      sm: "contents",
                      xl: "contents",
                    },
                  }}
                >
                  {routes.map((page) => (
                    <Grid key={page.id} item xl={1} md={2} sm={2} xs={1}>
                      <Button
                        id={page.name + "-btn"}
                        style={{ color: "#000000" }}
                        sx={{ my: 2, color: "white", display: "block" }}
                        onClick={() => redirectToPage(page.path)}
                      >
                        {page.name}
                      </Button>
                    </Grid>
                  ))}
                </Box>
              </Grid>
            </Grid>
            <Grid item xl={4} md={4} sm={4} xs={screenIsXS ? 10 : 4}>
              <Grid
                container
                direction={screenIsXS ? "row" : "column"}
                justifyContent="space-around"
                alignItems="flex-end"
                sx={{ marginBottom: { sm: "6px" }, marginLeft: { xs: "8px" } }}
              >
                <Grid
                  item
                  xl={12}
                  md={12}
                  sm={12}
                  xs={screenIsXS ? 6 : 12}
                  sx={{ marginTop: { xl: "8px", md: "8px", sm: 0 } }}
                >
                  <ButtonGroup
                    size="small"
                    color="primary"
                    variant="text"
                    aria-label="text button group"
                    className="buttonGroup"
                    sx={{
                      "& .MuiButtonGroup-grouped": {
                        borderRight: "none !important",
                      },
                    }}
                  >
                    <Button
                      id="register-page-btn"
                      color="primary"
                      sx={{
                        fontSize: "1rem",
                        textTransform: "none",
                      }}
                      onClick={() => history.push("/register")}
                    >
                      SignIn
                    </Button>
                    <Button
                      id="login-page-btn"
                      color="primary"
                      sx={{
                        fontSize: "1rem",
                        textTransform: "none",
                      }}
                      onClick={() => history.push("/login")}
                    >
                      Login
                    </Button>
                  </ButtonGroup>
                </Grid>

                <Grid item xl={12} md={12} sm={12} xs={screenIsXS ? 6 : 12}>
                  <Box
                    sx={{
                      display: {
                        xs: "contents",
                        md: "contents",
                        sm: "contents",
                        xl: "contents",
                      },
                    }}
                  >
                    <Button
                      id="cart-btn"
                      aria-describedby={id}
                      variant="outlined"
                      startIcon={<ShoppingCartIcon />}
                      color="primary"
                      onClick={handleClick}
                      className="cartButton"
                    >
                      {cartProductsData &&
                        cartProductsData.data &&
                        cartProductsData.data.length}{" "}
                      Items
                    </Button>
                    <Popover
                      id={id}
                      open={open}
                      anchorEl={anchorEl}
                      onClose={handleClose}
                      anchorOrigin={{
                        vertical: "bottom",
                        horizontal: "right",
                      }}
                      transformOrigin={{
                        vertical: "top",
                        horizontal: "right",
                      }}
                      PaperProps={{
                        style: {
                          width: "30%",
                          height: "80%",
                          padding: 0,
                          margin: 0,
                          backgroundColor: "#efefef",
                          zIndex: "inherit",
                        },
                      }}
                    >
                      <Typography component="div">
                        <Cart />
                      </Typography>
                    </Popover>
                  </Box>
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        </Toolbar>
      </Container>
    </AppBar>
  );
};
