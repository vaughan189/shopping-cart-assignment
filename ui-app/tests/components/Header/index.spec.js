import React from "react";
import reactRouterDom from "react-router-dom";
import { shallow } from "enzyme";
import { Header } from "../../../src/components/Header";
import { useMediaQuery } from "@mui/material";

import { withHooks } from "jest-react-hooks-shallow";

jest.mock("react-router-dom");

const pushMock = jest.fn();
reactRouterDom.useHistory = jest.fn().mockReturnValue({ push: pushMock });

const anchorMock = jest.fn();

jest.mock("@mui/material", () => ({
  ...jest.requireActual("@mui/material"),
  useMediaQuery: jest.fn().mockReturnValue(false),
}));

describe("Header component", () => {
  it("should render the Header component correctly", () => {
    withHooks(() => {
      const component = shallow(<Header debug />);
      expect(component).toMatchSnapshot();
    });
  });

  it("should trigger the show cart function on button click", () => {
    const component = shallow(<Header debug />);
    component
      .find("#cart-btn")
      .simulate("click", { currentTarget: anchorMock });
    expect(anchorMock).toHaveBeenCalled();
  });

  it("should trigger the history push cart function on button click", () => {
    useMediaQuery.mockReturnValue(true);
    const component = shallow(<Header debug />);
    component
      .find("#cart-btn")
      .simulate("click", { currentTarget: anchorMock });
    expect(anchorMock).toHaveBeenCalled();
  });

  it("should trigger the history push when clicked on home button", () => {
    const component = shallow(<Header debug />);
    component.find("#Home-btn").simulate("click");
    expect(pushMock).toHaveBeenCalled();
  });
  it("should trigger the history push when clicked on home icon button", () => {
    const component = shallow(<Header debug />);
    component.find("#home-icon-btn").simulate("click");
    expect(pushMock).toHaveBeenCalled();
  });
  it("should trigger the history push when clicked on register button", () => {
    const component = shallow(<Header debug />);
    component.find("#register-page-btn").simulate("click");
    expect(pushMock).toHaveBeenCalled();
  });
  it("should trigger the history push when clicked on login button", () => {
    const component = shallow(<Header debug />);
    component.find("#login-page-btn").simulate("click");
    expect(pushMock).toHaveBeenCalled();
  });
});
