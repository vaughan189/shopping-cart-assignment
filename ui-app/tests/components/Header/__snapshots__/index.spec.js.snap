// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the Header component correctly 1`] = `
<ForwardRef(AppBar)
  position="sticky"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "height": "10%",
    }
  }
>
  <ForwardRef(Container)
    maxWidth="xl"
  >
    <ForwardRef(Toolbar)
      disableGutters={true}
    >
      <ForwardRef(Grid)
        container={true}
        direction="row"
      >
        <ForwardRef(Grid)
          item={true}
          md={2}
          sm={2}
          xl={2}
          xs={2}
        >
          <img
            alt="Sabka Bazaar Logo"
            id="home-icon-btn"
            onClick={[Function]}
            src={Object {}}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          />
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={6}
          sm={6}
          xl={6}
          xs="auto"
        >
          <ForwardRef(Grid)
            alignItems="center"
            container={true}
            direction="row"
            justifyContent="center"
          >
            <ForwardRef(Box)
              sx={
                Object {
                  "display": Object {
                    "md": "contents",
                    "sm": "contents",
                    "xl": "contents",
                    "xs": "none",
                  },
                }
              }
            >
              <ForwardRef(Grid)
                item={true}
                key="1"
                md={2}
                sm={2}
                xl={1}
                xs={1}
              >
                <ForwardRef(Button)
                  id="Home-btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "#000000",
                    }
                  }
                  sx={
                    Object {
                      "color": "white",
                      "display": "block",
                      "my": 2,
                    }
                  }
                >
                  Home
                </ForwardRef(Button)>
              </ForwardRef(Grid)>
              <ForwardRef(Grid)
                item={true}
                key="2"
                md={2}
                sm={2}
                xl={1}
                xs={1}
              >
                <ForwardRef(Button)
                  id="Products-btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "#000000",
                    }
                  }
                  sx={
                    Object {
                      "color": "white",
                      "display": "block",
                      "my": 2,
                    }
                  }
                >
                  Products
                </ForwardRef(Button)>
              </ForwardRef(Grid)>
            </ForwardRef(Box)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={4}
          sm={4}
          xl={4}
          xs={4}
        >
          <ForwardRef(Grid)
            alignItems="flex-end"
            container={true}
            direction="column"
            justifyContent="space-around"
            sx={
              Object {
                "marginBottom": Object {
                  "sm": "6px",
                },
                "marginLeft": Object {
                  "xs": "8px",
                },
              }
            }
          >
            <ForwardRef(Grid)
              item={true}
              md={12}
              sm={12}
              sx={
                Object {
                  "marginTop": Object {
                    "md": "8px",
                    "sm": 0,
                    "xl": "8px",
                  },
                }
              }
              xl={12}
              xs={12}
            >
              <ForwardRef(ButtonGroup)
                aria-label="text button group"
                className="buttonGroup"
                color="primary"
                size="small"
                sx={
                  Object {
                    "& .MuiButtonGroup-grouped": Object {
                      "borderRight": "none !important",
                    },
                  }
                }
                variant="text"
              >
                <ForwardRef(Button)
                  color="primary"
                  id="register-page-btn"
                  onClick={[Function]}
                  sx={
                    Object {
                      "fontSize": "1rem",
                      "textTransform": "none",
                    }
                  }
                >
                  SignIn
                </ForwardRef(Button)>
                <ForwardRef(Button)
                  color="primary"
                  id="login-page-btn"
                  onClick={[Function]}
                  sx={
                    Object {
                      "fontSize": "1rem",
                      "textTransform": "none",
                    }
                  }
                >
                  Login
                </ForwardRef(Button)>
              </ForwardRef(ButtonGroup)>
            </ForwardRef(Grid)>
            <ForwardRef(Grid)
              item={true}
              md={12}
              sm={12}
              xl={12}
              xs={12}
            >
              <ForwardRef(Box)
                sx={
                  Object {
                    "display": Object {
                      "md": "contents",
                      "sm": "contents",
                      "xl": "contents",
                      "xs": "contents",
                    },
                  }
                }
              >
                <ForwardRef(Button)
                  className="cartButton"
                  color="primary"
                  id="cart-btn"
                  onClick={[Function]}
                  startIcon={<Memo />}
                  variant="outlined"
                >
                   
                  Items
                </ForwardRef(Button)>
                <ForwardRef(Popover)
                  PaperProps={
                    Object {
                      "style": Object {
                        "backgroundColor": "#efefef",
                        "height": "80%",
                        "margin": 0,
                        "padding": 0,
                        "width": "30%",
                        "zIndex": "inherit",
                      },
                    }
                  }
                  anchorEl={null}
                  anchorOrigin={
                    Object {
                      "horizontal": "right",
                      "vertical": "bottom",
                    }
                  }
                  onClose={[Function]}
                  open={false}
                  transformOrigin={
                    Object {
                      "horizontal": "right",
                      "vertical": "top",
                    }
                  }
                >
                  <ForwardRef(Typography)
                    component="div"
                  >
                    <Cart />
                  </ForwardRef(Typography)>
                </ForwardRef(Popover)>
              </ForwardRef(Box)>
            </ForwardRef(Grid)>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Toolbar)>
  </ForwardRef(Container)>
</ForwardRef(AppBar)>
`;
